app/models/apple/show.rb

```diff
@@ -147,9 +148,26 @@ def get_show
       self.class.get_show(api, apple_id)
     end
 
+    def sort_by_episode_properties(eps)
+      eps.sort_by do |e|
+        [
+          e.deleted_at.nil? ? 1 : -1,
+          e.published_at.present? ? 1 : -1,
+          e.published_at || e.created_at,
+          e.created_at
```

Comment:
Just seeing if I understand this - so with the `.reverse` this is sorting by:

1. created_at asc
2. published_at asc
3. non-blank published at, followed by blank (drafts)
4. non-deleted, followed by deleted

What's the reasoning for that?  So we always sync the oldest episodes first?

What happens in the case where you delete an episode, then create/publish a new version of it with the same <guid>?  Is this sorting so we always process the deleted one first?

Reply:
Update: i read this wrong.  The `reverse` is on the sort_by, not on the array.

But probably still worth a comment explaining why which of these is most important.